January 2026 rollup fixes and improvements

Cyanoxygen opened the pull request:
EROFS support
============

A new environment variable `FSTYPE` is added, to allow specifying filesystem type when archiving the layers.

Currently it supports both `squashfs` and `erofs`.

Submenu for choosing various GPU drivers
===================================

We now ship more GPU drivers by default, and NVIDIA released `nvidia-open` for Turing+ (RTX 20XX Series) GPUs, making the GPU driver more diverse. It's time to provide a submenu for choosing the GPU being loaded and installed.

Cyanoxygen committed:
dracut: support multiple filesystem backends

Cyanoxygen committed:
chrore: support EROFS

Cyanoxygen committed:
templates/livekit: use sysusers.d to create the live user

By using sysusers.d, we can also work around the overridden /etc/passwd
when nvidia driver is overlayed on top of LiveKit, which contains a new
user nvndia-persistentd.

Cyanoxygen committed:
templates/livekit: adjust systemd presets

* Point display-manager.service to lightdm.service.
* Disable ldconfig.service, which takes a significant amount of time.

Cyanoxygen committed:
lib.bash: move everything from gen-installer to here

Cyanoxygen committed:
gen-recipe.py: move to helpers/

Cyanoxygen committed:
livekit: generate initranfs images directly to output directory

Cyanoxygen committed:
scripts/cleanup.sh: preserve ATM topics state; remove preset stamp

Removal of this stamp file will cause systemd to reapply system presets
during the first boot, which happens to be the case of booting the live
environment.

Cyanoxygen committed:
mklive: add overlays definition

Cyanoxygen committed:
mklive: add sysroots definition

Cyanoxygen committed:
mklive: add target sysroots definition and loader configuration

Cyanoxygen committed:
templates: use desktop-drivers.sh to handle initramfs generation

Cyanoxygen committed:
aosc-mklive: new overhauled script

* Read overlay, sysroot and target definition files instead of hard
  coding them.
* Simplify some generation logic.
* Use a standard getopt based command line handling.
* Move some post processing logic to separate scripts.

Cyanoxygen committed:
grub: add submenu for graphics options; various tweaks

* Add SYSTEMD_SULOGIN_FORCE=1 to allow initrd to log in when emergency
  mode is triggered.
* Use a date based volume ID instead of the constant LiveKit.

Cyanoxygen committed:
templates/desktop.sh: install packages from the mirror set in cmdline

Cyanoxygen committed:
dracut: update livekit-loader

Cyanoxygen committed:
boot/boot.cfg: add SYSTEMD_SULOGIN_FORCE=1 to allow shell access

Cyanoxygen committed:
mklive: remove old scripts

Cyanoxygen committed:
grub: update PO template file and translation catalogs

Cyanoxygen committed:
templates/desktop: disable ldconfig during startup

Cyanoxygen committed:
overlays: rename DEPS -> OVERLAY_DEPS

Cyanoxygen committed:
loongarch64: installer: remove desktop-loonggpu; add livekit

Cyanoxygen committed:
targets/installer: add livekit

Cyanoxygen committed:
grub: quirks: skip acquiring DMI data for mips64el-efi

Cyanoxygen committed:
grub: quirks: skip serial console check for "D920S10"

The machine freezes when checking for available serial consoles on this
board. Another broken firmware implementation ...

Cyanoxygen committed:
lib: force mount(8) to use mount(2) instead of fsconfig(2)

fsconfig(2) is used to configure the mount context, i. e. setting the
source, destination and mount options. However, the maximum length of
the string is 256, which is insufficient for the lowerdir= option of the
overlayfs. mount(2) allows PAGE_SIZE - 1 in length for the entire option
string. We won't hit this limit in the forceable future.

Cyanoxygen committed:
overlays/desktop-common: add qt-6

Cyanoxygen committed:
installer: make nvidia overlays depend on nvidia-common

Now that we worked around the limitation of fsconfig, we can reintroduce
the nvidia-common overlay, further reducing the overall size of the
installation media.

Cyanoxygen committed:
iso-post-process: update SHIM and signed GRUB

Cyanoxygen committed:
iso-post-process: update Secure Boot disablement prompt

Cyanoxygen committed:
dracut: update for libmount workaround

Cyanoxygen committed:
grub: make serial check opt-out by default

It causes too many problems and serial connections are seldomly used.

Cyanoxygen committed:
targets/livekit: remove POSTPROC

It is merged to livekit-base-postprocess.bash.

Cyanoxygen committed:
lib: make sure ISODIR/boot exists before creating initramfs image

Cyanoxygen committed:
templates/livekit: disable fc-cache.service

Cyanoxygen committed:
templates/livekit: override template files instead of instances

Also enable autologin for getty@ instances, in case of kmscon failed to
start.

Cyanoxygen committed:
aosc-mklive: install memtest86+ binaries from the container

This makes sure we have correct binaries when we are generating the
installer image from a different architecture.

Cyanoxygen committed:
grub: clean up drivers submenu

Removes loongarch64 specific options as there's no need to split out a
driver layer for loongarch64.

Cyanoxygen committed:
templates: disable fc-cache on startup; pre-generate caches

Cyanoxygen committed:
gen-recipe.py: make base distribution appear last on the list

Cyanoxygen committed:
iso-postprocess: add link for details

Cyanoxygen committed:
installer: add driver overlay for Glenfly Arise 1000/2000

MingcongBai commented on helpers/recipe.ini:
```diff
@@ -0,0 +1,66 @@
+name-tr=desktop-name
+dir-name=desktop
+retro=false
+desc=A full-featured desktop experience.
```
Full-featured desktop system.

MingcongBai commented on helpers/recipe.ini:
```diff
@@ -0,0 +1,66 @@
+name-tr=desktop+nvidia-name
+dir-name=desktop-nvidia
+retro=false
+desc=A full-featured desktop experience (with proprietary NVIDIA driver).
```
Full-featured desktop system (with drivers for NVIDIA GPUs up to RTX 20 series).

MingcongBai commented on helpers/recipe.ini:
```diff
@@ -0,0 +1,66 @@
+desc-tr=desktop-description
+
+[desktop-nvidia]
+name=Desktop (with NVIDIA GPU up to RTX 20 Series)
```
for NVIDIA GPUs up to RTX 20 series

MingcongBai commented on helpers/recipe.ini:
```diff
@@ -0,0 +1,66 @@
+desc-tr=desktop+nvidia-description
+
+[desktop-nvidia-open]
+name=Desktop (with NVIDIA GPU since RTX 20 Series)
```
for NVIDIA GPUs since RTX 20 series

MingcongBai commented on helpers/recipe.ini:
```diff
@@ -0,0 +1,66 @@
+name-tr=desktop+nvidia-open-name
+dir-name=desktop-nvidia-open
+retro=false
+desc=A full-featured desktop experience (with nvidia-open driver).
```
Full-featured desktop system (with drivers for NVIDIA GPUs since RTX 20 series).

MingcongBai commented on helpers/recipe.ini:
```diff
@@ -0,0 +1,66 @@
+name-tr=desktop+cx4-name
+dir-name=desktop-cx4
+retro=false
+desc=A full-featured desktop experience (with Zhaoxin C-1080/1190 graphics driver).
```
Ditto

MingcongBai replied:
with drivers for Zhaoxin C-1080/1190 GPUs

MingcongBai commented on helpers/recipe.ini:
```diff
@@ -0,0 +1,66 @@
+desc-tr=desktop+nvidia-open-description
+
+[desktop-cx4]
+name=Desktop (with Zhaoxin C-1080/1190 integrated graphics)
```
with => for

MingcongBai replied:
integrated ... GPUs

MingcongBai commented on helpers/recipe.ini:
```diff
@@ -0,0 +1,66 @@
+name-tr=desktop+zhaoxin
+dir-name=desktop-zhaoxin
+retro=false
+desc=A full-featured desktop experience (with Zhaoxin C-960 graphics driver).
```
Ditto; with drivers for integrated Zhaoxin C-960 GPUs

MingcongBai commented on helpers/recipe.ini:
```diff
@@ -0,0 +1,66 @@
+desc-tr=desktop+cx4-description
+
+[desktop-zhaoxin]
+name=Desktop (with Zhaoxin C-960 integrated graphics)
```
for; integrated ... GPUs

MingcongBai commented on helpers/recipe.ini:
```diff
@@ -0,0 +1,66 @@
+desc-tr=desktop+zhaoxin-description
+
+[desktop-arise]
+name=Desktop (with Glenfly Arise 1000/2000 graphics)
```
Ditto for all below.

- graphics => GPUs
- for... with...

MingcongBai commented on helpers/recipe.ini:
```diff
@@ -0,0 +1,66 @@
+desc=A minimal system with a basic set of utilities.
+desc-tr=base-description
+
+[desktop-latx]
```
Drop this.

MingcongBai committed:
templates/livekit: update background for jan-2026

Cyanoxygen committed:
lib.bash: remove unused pushd in pack_squashfs

Cyanoxygen committed:
chore: update strings in the recipe and GRUB configuration

Cyanoxygen committed:
templates/desktop-drivers: fix the name of default_config()

Cyanoxygen committed:
recipe.ini: fix translation key for desktop-zhaoxin

Cyanoxygen committed:
recipe: add translation support

* Add gitignore to ignore backup PO files in GRUB menu translations.

Cyanoxygen committed:
grub: update PO template and translation catalog files

Cyanoxygen committed:
scripts/iso-pre-process: generate recipe translation file

MingcongBai committed:
helpers/i18n: add localised recipe data

Cyanoxygen committed:
helpers/gen-recipe: add loonggpu

Cyanoxygen committed:
chore: add loonggpu for loongarch64

MingcongBai committed:
helper/i18n: improve wording and fix information on NVIDIA recipes

MingcongBai committed:
boot: update GRUB options and translation

MingcongBai committed:
helpers/i18n/zh-TW.ini: 臺式 => 桌上型

Cyanoxygen committed:
sysroots: use nvidia+580; remove nvidia-common

Since we are now using nvidia+580, there will be a significant
difference between the dependencies of the two drivers.

Cyanoxygen committed:
targets/installer: add loonggpu; remove nvidia-common

MingcongBai committed:
boot: helpers: 整合式圖形處理器 => 內建顯示卡

Suggested-by: Kaiyang Wu <origincode@aosc.io>

Cyanoxygen committed:
grub: boot with loonggpu driver by default on loongarch64

Cyanoxygen committed:
overlays: remove nvidia-common overlay for nvidia drivers

MingcongBai committed:
boot: helpers: refine wording for Chinese localisation

Cyanoxygen committed:
lib: wait for the lock file after terminating the container

Cyanoxygen committed:
aosc-mklive: support mirror URL override using REPO

* Improve help texts.

Cyanoxygen committed:
targets/installer: fix typo in livekit-nvidia for arm64

Cyanoxygen committed:
grub: fix typo: grub_gpu -> grub_cpu; update menu entry IDs

Cyanoxygen committed:
lib: install_overlay: avoid checking for DBus

This causes problems on QEMU user emulation, which will always timeout.

Cyanoxygen committed:
aosc-mklive: remove stub command from pre- and post-processing cmds

Cyanoxygen committed:
scripts/livekit: extract dracut module from tarball if not using mklive

Cyanoxygen committed:
grub: update memtest filenames for livekit

MingcongBai reviewed (approved):
Tested good on all primary architectures.
